main/src/ca/uwaterloo/flix/language/phase/typer/TypeReduction2.scala

```diff
@@ -0,0 +1,352 @@
+            case (TypeConstructor.Float64, TypeConstructor.Unit) => classOf[java.util.function.DoubleConsumer]
+            case (TypeConstructor.Float64, TypeConstructor.Bool) => classOf[java.util.function.DoublePredicate]
+            case (TypeConstructor.Float64, TypeConstructor.Float64) => classOf[java.util.function.DoubleUnaryOperator]
+            case (TypeConstructor.Float64, TypeConstructor.Native(obj)) if obj == classOf[Object] => classOf[java.util.function.DoubleFunction[Object]]
+            case (TypeConstructor.Int64, TypeConstructor.Unit) => classOf[java.util.function.LongConsumer]
+            case (TypeConstructor.Int64, TypeConstructor.Bool) => classOf[java.util.function.LongPredicate]
+            case (TypeConstructor.Int64, TypeConstructor.Int64) => classOf[java.util.function.LongUnaryOperator]
+            case (TypeConstructor.Int64, TypeConstructor.Native(obj)) if obj == classOf[Object] => classOf[java.util.function.LongFunction[Object]]
+            case (TypeConstructor.Native(obj), TypeConstructor.Unit) if obj == classOf[Object] => classOf[java.util.function.Consumer[Object]]
+            case (TypeConstructor.Native(obj), TypeConstructor.Bool) if obj == classOf[Object] => classOf[java.util.function.Predicate[Object]]
+            case _ => classOf[Object] // default
+          }
+        case _ => classOf[Object] // default
+      }
+    case _ => classOf[Object] // default
+  }
+
+  /** Tries to find a field of `thisObj` with the name `fieldName`. */
+  def lookupField(thisObj: Type, fieldName: String): JavaFieldResolution = {
+    val typeIsKnown = isKnown(thisObj)
+    if (!typeIsKnown) return JavaFieldResolution.UnresolvedTypes
+    val opt = for {
+      clazz <- Type.classFromFlixType(thisObj)
+      field <- JvmUtils.getField(clazz, fieldName, static = false)
+    } yield JavaFieldResolution.Resolved(field)
+    opt.getOrElse(JavaFieldResolution.NotFound)
+  }
+
+  /** Returns `true` if type is resolved enough for Java resolution. */
+  private def isKnown(tpe: Type): Boolean = tpe match {
+    case Type.Var(_, _) if tpe.kind == Kind.Eff => true
+    case Type.Var(_, _) => false
+    case Type.Cst(_, _) => true
+    case Type.JvmToType(_, _) => false
+    case Type.JvmToEff(_, _) => false
+    case Type.UnresolvedJvmType(_, _) => false
+    case Type.Apply(t1, t2, _) => isKnown(t1) && isKnown(t2)
+    case Type.Alias(_, _, t, _) => isKnown(t)
+    case Type.AssocType(_, _, _, _) => false
+  }
+
+  /** A lookup result of a Java field. */
+  sealed trait JavaFieldResolution
+
+  object JavaFieldResolution {
+
+    /** One matching field. */
+    case class Resolved(field: Field) extends JavaFieldResolution
+
+    /** No matching field. */
+    case object NotFound extends JavaFieldResolution
+
+    /**
+      * The types of the lookup are not resolved enough to decide
+      * (it contains type variables, associated types, etc.).
```

Comment:
it -> they
drop parens
Fix grammar and formatting.

Same comment applies to JavaMethodRes